_posts/2020-01-10-ios-logout-button-at-last.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: epic
+title: "At Last: A Log Out Button"
+date: "2020-01-14"
+author: [ash]
+categories: [ios, tech debt, hackathon]
+---
+
+In 2013, Artsy shipped the first version of our iOS app. Typical for an early-day startup, the app was a "minimum
+viable product" (with a big emphasis on "minimum"). One of the features that didn't make the cut was something you
+expect to see in most apps: a log out button.
+
+When I joined Artsy a year later, there was still no log out button. And there would be no log out button for
+another six years, until today.
+
+I want to talk about this quirk of our app, from both a product and technical perspective. I also want to talk
+about how we finally managed to prioritize this kinda weird feature request (spoilers: it was our company-wide
+hackathon). Let's go!
+
+<!-- more -->
+
+When I say that our app doesn't have a log out button, that's a bit of a fib: it _does_ have a log out button... in
+the admin-only debug menu. The reason why this isn't a user-facing feature is that the final step of this
+admin-only log out feature is a call to `exit(0)`, effectively crashing the app. That's _one_ way to make sure that
+user-specific state doesn't pollute your app's runtime, but of course it would be ridiculous for a user-facing log
+out button to crash the app. The only other way to log out was to uninstall the app, which is _not_ something we
+want to encourage users to do.
+
+So Artsy staff could log out of the app, but our normal users couldn't. This quirk was acceptable in the early days
+of our app, but as the years wore on, it became less of a quirk and more of a product limitation. Even three months
+after creating our new Mobile Experience team, we hadn't yet prioritized this feature.
+
+Why? Well, to be honest, we had – and still have, [we're hiring by the way](http://artsy.net/jobs) – a lot of work
+to do to improve the app. Among the high-impact work we've been shipping, the log out button never made the cut.
+Why is that? Well, it's actually quite complicated. And why is it complicated? Well for that, we need to step back
+and think about software development and requirements gathering.
```

Comment:
Something I see a lot of in this post is `Why? Well, ...`, both in this section and elsewhere. I know you're doing this as an interrogative-style, 5-whys-ish post, and I really like the conversational tone, but it can make following the train of thought a little difficult at points. Couple different ideas there (feel free to mix and match as you see fit):

- Add some kind of formatting and explicit callout of the "why"s in this post. For example, you could say something in the intro like 'When I looked at this problem, I said to myself, "This shouldn't be so hard. Why is it hard?" - this post will take you down the interrogative rabbit hole I fell into as I asked "why?" again and again.' and then you could bold or underline every "why" in the post
- Cut the `Well` out of most or all of those sentences
- Trim the number of "why"s a bit. E.g. I could see this paragraph saying something like "Why? Well, to be honest, we had – and still have, [we're hiring by the way](http://artsy.net/jobs) – a lot of work
to do to improve the app. Adding a log-out button was a low-priority, high-complexity issue that just never made it beyond sprint planning. \n\n But wait, why is it hard to add a log-out button? For that, we need to step back
and think about software development and requirements gathering."
- Breaking things up more would also work. When you go into a "why section" like this one, you could put every why on a newline - could be a nice way to break up the piece as a whole. You have some explanation, some exposition, and then more "why"s to bring us deeper into the problem

Hope that helps!